src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -538,8 +539,18 @@ def update_file_service_properties(cmd, instance, enable_delete_retention=None,
     # TODO: remove it when server side return null not 0 for days
     if instance.share_delete_retention_policy is not None and instance.share_delete_retention_policy.enabled is False:
         instance.share_delete_retention_policy.days = None
+    if instance.share_delete_retention_policy:
+        params['share_delete_retention_policy'] = instance.share_delete_retention_policy
 
-    return instance
+    # set protocol settings
+    if enable_smb_multichannel is not None:
+        instance.protocol_settings = cmd.get_models('ProtocolSettings')()
+        instance.protocol_settings.smb = cmd.get_models('SmbSetting')(
+            multichannel=cmd.get_models('Multichannel')(enabled=enable_smb_multichannel))
+    if instance.protocol_settings.smb.multichannel:
+        params['protocol_settings'] = instance.protocol_settings
```

Comment:
Why another `if` instead of setting `params['protocal_settings']` within `if enable_smb_multichannel is not None`?

If the origin instance settings is `True` for `protocol_settings.smb.multichannel` and customer don't need to change, shall we still set `params['protocal_settings']` here?

Reply:
This part code is copied from azure cli extension storage preview as you can see in 
https://github.com/Azure/azure-cli-extensions/blob/19b4eaa8f47bd0c4be5dcdd28acc39956b9bf5e2/src/storage-preview/azext_storage_preview/operations/account.py#L302-L308

I have such code because if previous `protocol_settings.smb.multichannel` is None, I will not add `protocol_setting` in params because for storage account type which doesn't support protocol setting will throw error in server side if we have schema `protocol_setting.smb` in params no matther it is None or non-None.
 For the storage account type which not support smb setting, SRP will still return with protocol_settings.smb although it is None.

And if it is true, resetting it with the same value will not cause any issue.